Media related interfaces A2DP

Ch4rg3r opened the pull request:
Added MediaEndpoint1
Added MediaTransport1
Added most A2Dp constants & properties

Ch4rg3r committed:
Media

Media Interfaces for sink

Ch4rg3r committed:
A2DP

Add A2DP properties

Ch4rg3r committed:
Fix A2DP

brookpatten commented:
Looking good, do you have an example of using these?  Like a demo app or something?

Ch4rg3r commented:
Not yet. it's pretty hard to convert all of these from Bluez docs. I used simple-endpoint example from docs (python). But i'm stuck on implementation of Endpoint.
In Python, they had created the class Endpoint, don't know why.

Basically, you need to get Bluez to work, and register the endpoint to the adapter with a new path like `"/whatever/endpoint"` or similar. Then it should work, as it will use the mediatransport for the stream.

Also i have no idea how to get connection to work. In this case, the App needs to wait for a Device to connect (no idea what i need of Bluez for that to work)

EDIT:
Of course this is used for the app to react as a Sink, you can stream Audio to it. But the other way round works mostly the same, except for the waiting for a connection.

brookpatten commented:
Cool.  If possible I'd like to have a working example before merging the PR.  I should have time to play with it some next week if you don't beat me to it.

Ch4rg3r commented:
EDIT:

![image](https://cloud.githubusercontent.com/assets/11695227/14704578/aec07fc6-07b5-11e6-915c-e35a618df58d.png)
[http://www.lightofdawn.org/wiki/wiki.cgi/BluezA2DP](Source)

I implemented everything needed like explained in above linked text. Added test to console, not yet tried it.

Ch4rg3r committed:
Fix MediaTransport1

enum -> static class with

Ch4rg3r committed:
Bluezutils

Added Bluezutils -> converted from Python example
Fixes

Ch4rg3r committed:
Fix A2DP.cs location

Delete A2Dp.cs from Mono.DBus (duplicate)

Ch4rg3r committed:
Fix Namespaces

Ch4rg3r committed:
Fixes

Ch4rg3r committed:
A2Dp Test Project

Added testproject (not yet tested)
Added AudioSink & AudioSource interfaces
Fixes

Ch4rg3r commented:
> Cool. If possible I'd like to have a working example before merging the PR. I should have time to play with it some next week if you don't beat me to it.

So i added a test project a while ago, but never was able to test it really. Now it got my RPi3 and i will rewrite it to check for connection to the adapter and endpoints...
